tests/Aspire.StackExchange.Redis.Tests/AspireRedisExtensionsTests.cs

```diff
@@ -223,4 +226,28 @@ public void MultipleRedisComponentsCanBeAdded(bool useKeyed)
         var healthCheckService = host.Services.GetRequiredService<HealthCheckService>();
         Assert.NotNull(healthCheckService);
     }
+
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "_profilingSessionProvider")]
+    static extern ref Func<ProfilingSession>? GetProfiler(ConnectionMultiplexer? @this);
+
+    [Fact]
+    public void KeyedServiceRedisInstrumentation()
+    {
+        var builder = Host.CreateEmptyApplicationBuilder(null);
+
+        builder.AddKeyedRedis("redis", settings =>
+        {
+            settings.ConnectionString = "localhost";
+            settings.Tracing = true;
+        }, options => { });
+        var host = builder.Build();
+
+        //This will add the instrumentations.
+        var tracerProvider = host.Services.GetRequiredService<TracerProvider>();
+
+        var connectionMultiplexer = host.Services.GetRequiredKeyedService<IConnectionMultiplexer>("redis");
+        var profiler = GetProfiler(connectionMultiplexer as ConnectionMultiplexer);
+
+        Assert.NotNull(profiler);
```

Comment:
Instead of this, can we do an operation and ensure traces happen?

Reply:
Look at the tests at the top of this file.

https://github.com/dotnet/aspire/blob/b7ef6613f08359a10746e2285505badf513e9fff/tests/Aspire.StackExchange.Redis.Tests/AspireRedisExtensionsTests.cs#L19-L22

When a Redis server can't be connected to, the test is skipped. When working on these components, we run a local docker container that the tests connect to, and the full suite of tests run.

See 

https://github.com/dotnet/aspire/blob/b7ef6613f08359a10746e2285505badf513e9fff/tests/Aspire.StackExchange.Redis.Tests/AspireRedisHelpers.cs#L12-L48

Reply:
Let's do both tests - the one you currently have, and a full one that is skipped if the connection isn't available.